panel/widgets/tables.py

```diff
@@ -176,34 +177,18 @@ def _get_column_definitions(self, col_names: List[str], df: pd.DataFrame) -> Lis
             col_kwargs = {}
             kind = data.dtype.kind
             editor: CellEditor
-            formatter: CellFormatter
+            formatter: CellFormatter | None = self.formatters.get(col)
             if kind == 'i':
-                formatter = NumberFormatter(text_align='right')
```

Comment:
Before this `formatter` was always created even though it would be overwritten further down if it existed in `self.formatter` .

So I have split this up into to two if-blocks one for `editor` and another for `formatter`